docs/3-v6-beta/4-flash-fill.md

```diff
@@ -1,9 +1,321 @@
+
+:::danger A note of caution
+_Unlike a typical flash loan, the repayment is in a different mint from the loan. As such, there is no easy way to ensure that the repayment amount is appropriate, do to take extra measures to minimize the surface area for exploits. For e.g. making the instruction permissioned to trusted admins or utilizing a price oracle etc+. The exact implementation is protocol specific and, hence, beyond the scope of this guide._
+:::
+
+**Program code example:**
+```rust
+#[event_cpi]
+#[derive(Accounts)]
+pub struct FlashFill<'info> {
+    admin: Signer<'info>,
+
+    #[account(mut)]
+    escrow: Account<'info, Escrow>,
+
+    input_mint: Account<'info, Mint>,
+    output_mint: Account<'info, Mint>,
+
+    /// the admin's input_mint account
+    #[account(
+        init_if_needed,
+        payer=admin,
+        associated_token::mint=input_mint,
+        associated_token::authority=admin,
+    )]
+    admin_input_token_account: Account<'info, TokenAccount>,
+
+    /// The account to borrow from
+    #[account(
+      mut,
+      associated_token::mint=input_mint,
+      associated_token::authority=escrow,
+    )]
+    escrow_input_token_account: Account<'info, TokenAccount>,
+
+    /// The account to repay to
+    #[account(
+        init_if_needed,
+        payer=admin,
+        associated_token::mint=output_mint,
+        associated_token::authority=escrow,
+    )]
+    escrow_output_token_account: Account<'info, TokenAccount>,
+
+    /// CHECK: Solana Instructions Sysvar
+    #[account(address = sysvar::instructions::ID)]
+    pub instructions_sysvar: UncheckedAccount<'info>,
+
+    system_program: Program<'info, System>,
+    token_program: Program<'info, Token>,
+    associated_token_program: Program<'info, AssociatedToken>,
+}
+
+pub fn flash_fill(ctx: Context<FlashFill>, amount: u64) -> Result<()> {
+    let instructions_sysvar = ctx.accounts.instructions_sysvar.to_account_info();
+
+    // check #1: make sure the current instruction isn't a cpi call
+    let current_idx = load_current_index_checked(&instructions_sysvar)? as usize;
+    let current_ixn = load_instruction_at_checked(current_idx, &instructions_sysvar)?;
+    require_keys_eq!(
+        current_ixn.program_id,
+        crate::ID
+    );
+
+    // check #2: check that the next instruction in the transaction performs a swap
+    let swap_ix_index = current_idx + 1;
+    if let Ok(jup_swap_ixn) = load_instruction_at_checked(swap_ix_index, &instructions_sysvar) {
+        // check #2a: check that it calls Jup v6 program
+        require_keys_eq!(
+            jup_swap_ixn.program_id,
+            Pubkey::from_str("JUP6LkbZbjS1jKKwapdHNy74zcZ3tLUZoi5QNyVTaV4").unwrap()
+        );
+
+        // check #2b: check that it calls Jup's swap instruction
+        let ixn_identifier = u64::from_be_bytes(jup_swap_ixn.data[..8].try_into().unwrap());
+        require!(
+            ixn_identifier == get_instruction_discriminator(&[b"global:shared_accounts_route"])
```

Comment:
we need to know what instruction when they put different params when getting the route

Reply:
https://github.com/jup-ag/space-station/commit/1fe458de94bef7b7476ddec2c55c0cc94b541eb2